dexter/src/main/java/com/karumi/dexter/MultiplePermissionListenerThreadDecorator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter;
+
+import com.karumi.dexter.listener.PermissionRequest;
+import com.karumi.dexter.listener.multi.MultiplePermissionsListener;
+import java.util.List;
+
+/**
+ * Decorator to execute the permission updates on a given thread specification
+ */
+final class MultiplePermissionListenerThreadDecorator implements MultiplePermissionsListener {
+
+  private final MultiplePermissionsListener listener;
+  private final Thread thread;
+
+  public MultiplePermissionListenerThreadDecorator(MultiplePermissionsListener listener,
```

Comment:
We should change the visibility of this constructor to ``default`` instead of ``public`` as we've done with the class signature.